src/components/__test__/GeneralButton.spec.tsx

```diff
@@ -0,0 +1,31 @@
+import React = require('react');
+import { render, screen, fireEvent } from '@testing-library/react';
+import GeneralButton from '../GeneralButton';
+
+describe('GeneralButton', () => {
+  it('renders corrrectly and has buttonText ADD RULE', () => {
+    const onClick = jest.fn();
+    render(
+      <GeneralButton className="test" buttonText="ADD RULE" onClick={onClick} />
+    );
+    expect(screen.getByText(/ADD RULE/)).toBeInTheDocument();
+  });
+
+  it('renders corrrectly and has buttonText DELETE', () => {
```

Comment:
This test is very similar to the previous test we have already tested that the button renders and has whichever text it's given so this might be unnecessary. So perhaps give the previous test a different name say `renders correctly and contains given button text`, this may not be the best way to put it, but something along those lines